test/e2e/status_test.go

```diff
@@ -0,0 +1,187 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer/pkg/cloudprovider"
+	operator "github.com/openshift/windows-machine-config-operator/pkg/apis/wmc/v1alpha1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// testStatusWhenSuccessful ensures that the status matches the expected state of the operator when the operator should
+// work correctly
+func testStatusWhenSuccessful(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+	defer testCtx.cleanup()
+
+	// wait until reconcile is complete
+	_, err = testCtx.waitForCondition(operator.Reconciling, corev1.ConditionFalse)
+	require.NoError(t, err, "error waiting for degraded condition")
+
+	// get WMC custom resource
+	wmc := &operator.WindowsMachineConfig{}
+	err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: wmcCRName,
+		Namespace: testCtx.namespace}, wmc)
+	require.NoError(t, err, "Could not retrieve instance of WMC")
+
+	assert.Equal(t, wmc.Spec.Replicas, wmc.Status.JoinedVMCount, "Num of nodes in status not equal to spec")
+
+	degraded := wmc.Status.GetWindowsMachineConfigCondition(operator.Degraded)
+	require.NotNil(t, degraded)
+	assert.Equal(t, corev1.ConditionFalse, degraded.Status, "Status shows operator degraded")
+}
+
+// testFailureSuite contains tests which involve invoking a reconcile failure. Requires gc.numberOfNodes to be at
+// least 1 to run.
+func testFailureSuite(t *testing.T) {
+	if gc.numberOfNodes < 1 {
+		t.Skip("testFailureSuite requires 1 or more nodes to run")
+	}
+	t.Run("VM provision failure", testVMProvisionFail)
+	t.Run("VM configuration failure", testVMConfigurationFail)
+}
+
+// cleanupWMC attempts to scale down the Windows nodes to 0 and delete the WMC object
+func (tc *testContext) cleanupWMC() {
+	wmc := &operator.WindowsMachineConfig{}
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: wmcCRName, Namespace: tc.namespace},
+		wmc)
+	if err != nil {
+		log.Printf("could not get WMC for cleanup: %v", err)
+		return
+	}
+
+	wmc.Spec.Replicas = 0
+	if err = framework.Global.Client.Update(context.TODO(), wmc); err != nil {
+		log.Printf("error updating WMC replica field: %v", err)
+	} else {
+		// if no error updating the WMC CR, wait for nodes to hit 0
+		if err = tc.waitForWindowsNodes(0, false); err != nil {
+			log.Printf("error waiting for Windows nodes to scale down %v", err)
+		}
+	}
+	if err = framework.Global.Client.Delete(context.TODO(), wmc); err != nil {
+		log.Printf("error removing WMC CR during cleanup: %v", err)
+	}
```

Comment:
I don't follow, this is within the cleanup function already?

Reply:
Sorry I meant the testctx [cleanup()](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/main_test.go#L88) function. But that might require making the CRs we are creating as part of the `testctx`. This would be to ensure that they are always cleaned up.

Reply:
Its not safe to do that currently. Some tests within the creation suite call testctx.Cleanup(), adding this code there would break the deletion suite. Ideally I think we refactor the creation and deletion suite into one test with multiple subtests and use a single testctx for that.

Reply:
I am fine leaving it as it is, however

>Ideally I think we refactor the creation and deletion suite into one test with multiple subtests and use a single testctx for that.

Nack on that. Ideally I should be able to call individually tests but that is a bit hard for us to do. The SDK pattern is also to use separate test contexts and not share them.